### PR TITLE
Add qasm rep delay

### PR DESCRIPTION
**Pull request by zachschoenfeld33**
## Summary

Add `rep_delay` in QASM path, analogous to pulse.

### Details and comments
- Can now specify `rep_delay` in QASM qobj
- backend rep delay flags (`dynamic_reprate_enabled`, `default_rep_delay`, `rep_delay_range`) now included in `QasmBackendConfiguration`

### CODE REVIEW

**Review by lcapelluto — DISMISSED** [review]
LGTM.

File: qiskit/providers/models/backendconfiguration.py [hunk 1]
```diff
@@ -403,6 +422,8 @@ class BackendConfiguration(QasmBackendConfiguration):
 class PulseBackendConfiguration(QasmBackendConfiguration):
     """Static configuration state for an OpenPulse enabled backend. This contains information
     about the set up of the device which can be useful for building Pulse programs.
+
+    Inherits from Qasm backend configuration.
```
[the remark this answers is not in the document]
Comment (zachschoenfeld33) [reply qiskit/providers/models/backendconfiguration.py]: Addressed in https://github.com/Qiskit/qiskit-terra/pull/5048/commits/6e1bc7c296dd5928c2b6008b8e093c9b9df2dd32.

**Review by taalexander — CHANGES_REQUESTED** [review]

File: qiskit/qobj/qasm_qobj.py [hunk 2]
```diff
@@ -254,6 +254,11 @@ def __init__(self, shots=None, max_credits=None, seed_simulator=None,
             n_qubits (int): The number of qubits on the device
             pulse_library (list): List of :class:`PulseLibraryItem`.
             calibrations (QasmExperimentCalibrations): Information required for Pulse gates.
+            rep_delay (float): Delay between programs in sec. Only supported on certain
```
[the remark this answers is not in the document]
Comment (zachschoenfeld33) [reply qiskit/qobj/qasm_qobj.py:257]: Addressed in https://github.com/Qiskit/qiskit-terra/pull/5048/commits/37f21b73c1844f3d991ed3f57bd4dee89627832d.

**Review by taalexander — DISMISSED** [review]

**Comment by zachschoenfeld33** [comment]
Added in https://github.com/Qiskit/qiskit-terra/pull/5048/commits/cc649b2579e79ba6146a1fab0eab732ed805b0e5 fix for issue cited by @ajavadia where warning occurs for dynamic rep rate not on even when `rep_delay` is not entered in assemble.

File: qiskit/schemas/backend_configuration_schema.json [hunk 3]
```diff
@@ -140,6 +140,22 @@
                         "type": "string",
```
Comment (taalexander) [inline qiskit/schemas/backend_configuration_schema.json:140]: The schema version needs a minor bump.

File: qiskit/schemas/qobj_schema.json [hunk 4]
```diff
@@ -1116,6 +1116,11 @@
                         "pulse_library": {
```
Comment (taalexander) [inline qiskit/schemas/qobj_schema.json:1116]: Schema needs a minor version bump.

**Review by taalexander — CHANGES_REQUESTED** [review]

**Review by taalexander — APPROVED** [review]

**Pull request merged**